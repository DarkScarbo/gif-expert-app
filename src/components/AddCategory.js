import React, { useState } from 'react';
import Proptypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => setInputValue(e.target.value);
  const handleSubmit = (e) => {
    e.preventDefault();
    setCategories(ctgs => [inputValue, ...ctgs]);
    setInputValue("")
  };

    return (
        <form onSubmit={handleSubmit}>
          <input
              type="text"
              value={inputValue}
              onChange={handleInputChange}
              required
          /> 
        </form>
    )
}

AddCategory.propTypes = {
  setCategories: Proptypes.func.isRequired
 }