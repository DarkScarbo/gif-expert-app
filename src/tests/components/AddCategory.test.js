import React from 'react'
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';
const { shallow } = require("enzyme")

describe('Tests in <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;
    const value = 'Rupaul'; 

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories} />);
    });
   
    test('should show the component correctly', () => {  
        expect(wrapper).toMatchSnapshot();
    })

    test('should change input box', () => {  
        const input = wrapper.find('input');     
        input.simulate('change', { target: { value } }); 

        const inputValue = wrapper.find('input').prop('value');
        expect(inputValue).toBe(value)
    })
    
    test('should not submit an empty input', () => {  
        wrapper.find('form').simulate('submit', { preventDefault: () => {} }); 
        
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('should call setCategories and reset input', () => {

        const input = wrapper.find('input');
        input.simulate('change', { target: { value } });

        const form = wrapper.find('form'); 
        form.simulate('submit', { preventDefault: () => {} }); 

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe("");
    })
    
})
