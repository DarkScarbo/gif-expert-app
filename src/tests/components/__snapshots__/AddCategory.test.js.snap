// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> should show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
</form>
`;
